### PR TITLE
Voice Signed Callbacks DevPreview

### PR DESCRIPTION
**Pull request by vshisterov**
Voice Signed Callbacks feature [Developer Preview] guide

### CODE REVIEW

**Comment by vshisterov** [comment]
Ready for review/adjustments, but please do not merge before the Support team reviews the guide (I'll remove the label once done).

**Comment by vshisterov** [comment]
We have a green light from the Support team, let's proceed with publishing!

File: _documentation/en/voice/voice-api/guides/signed-webhooks.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+<li style='margin-bottom:16px;'>activation can be done through Applications API request only,</li></ul>
+see more details below.
+
+## Activation
+
+Follow these steps to activate signed callbacks for your application.
+
+### Get Application
+
+Retrieve your application data with [Get an application](/api/application.v2#getApplication) HTTP request using [Postman](/tools/postman) or another HTTP client of your choice:
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/signed-webhooks.md:22]:
```suggestion
Retrieve your application data with a [Get an application](/api/application.v2#getApplication) HTTP request using [Postman](/tools/postman) or another HTTP client of your choice:
```

File: _documentation/en/voice/voice-api/guides/signed-webhooks.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    },
+    "_links": {
+        "self": {
+            "href": "/v2/applications/YOUR_APPLICATION_ID"
+        }
+    }
+}
+```
+
+> Unlike Voice API, Applications API uses [header-based API Key and Secret Authentication] (https://developer.nexmo.com/concepts/guides/authentication#header-based-api-key-and-secret-authentication), which means you should use [Base64](https://tools.ietf.org/html/rfc4648#section-4) encoded API key and secret joined by a colon in the `Authorization` header of the HTTP request.
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/signed-webhooks.md:63]:
```suggestion
> Unlike Voice API, the Applications API uses [header-based API Key and Secret Authentication] (https://developer.nexmo.com/concepts/guides/authentication#header-based-api-key-and-secret-authentication), which means you should use [Base64](https://tools.ietf.org/html/rfc4648#section-4) encoded API key and secret joined by a colon in the `Authorization` header of the HTTP request.
```

File: _documentation/en/voice/voice-api/guides/signed-webhooks.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+### Update Application
+
+Update your application with [Update an application](https://developer.nexmo.com/api/application.v2#updateApplication) HTTP request:
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/signed-webhooks.md:67]:
```suggestion
Update your application with a [Update an application](https://developer.nexmo.com/api/application.v2#updateApplication) HTTP request:
```

File: _documentation/en/voice/voice-api/guides/signed-webhooks.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+```http
+PUT https://api.nexmo.com/v2/applications/YOUR_APPLICATION_ID
+```
+
+using the response JSON from the previous step as the request body with the addition of `signed_callbacks` parameter:
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/signed-webhooks.md:73]:
```suggestion
Use the response JSON from the previous step as the request body with the addition of the `signed_callbacks` parameter:
```

File: _documentation/en/voice/voice-api/guides/signed-webhooks.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    },
+    "_links": {
+        "self": {
+            "href": "/v2/applications/YOUR_APPLICATION_ID"
+        }
+    }
+}
+```
+
+You can make a GET request from step one again to ensure the parameter is applied (it should be returned in the response).
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/signed-webhooks.md:109]:
```suggestion
You can make a `GET` request from step one again to ensure the parameter is applied (it should be returned in the response).
```

File: _documentation/en/voice/voice-api/guides/signed-webhooks.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+> Developer Preview limitation: if you change any parameter of your application via [Dashboard](https://dashboard.nexmo.com), the `signed_callbacks` parameter will be dropped, and the feature will be inactivated, so you have to go through the activation steps again to turn it back on.
+
+## Validating Signed Webhooks
+
+See [Signed webhooks](https://developer.nexmo.com/messages/concepts/signed-webhooks#validating-signed-webhooks) in the Messages API Concepts section. Voice signed callbacks have the same [JWT](https://jwt.io/) in the `Authorization` header, which your application is supposed to validate.
+
+## Supported Events and Scenarios
+
+At the moment, only the following events supported:
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/signed-webhooks.md:119]:
```suggestion
At the moment, only the following events are supported:
```

File: _documentation/en/voice/voice-api/guides/signed-webhooks.md [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+* [`answered`](/voice/voice-api/webhook-reference#answered)
+* [`completed`](/voice/voice-api/webhook-reference#completed)
+* [`busy`](/voice/voice-api/webhook-reference#busy)
+* [`cancelled`](/voice/voice-api/webhook-reference#cancelled)
+* [`unanswered`](/voice/voice-api/webhook-reference#unanswered)
+* [`rejected`](/voice/voice-api/webhook-reference#rejected)
+* [`failed`](/voice/voice-api/webhook-reference#failed)
+* [`record`](/voice/voice-api/webhook-reference#record)
+
+All the other events, as well as the initial [answer](/voice/voice-api/webhook-reference#answer-webhook) webhook, are still sent without signature (`Authorization` header).
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/signed-webhooks.md:132]:
```suggestion
All the other events, as well as the initial [answer](/voice/voice-api/webhook-reference#answer-webhook) webhook, are still sent without the signature (`Authorization` header).
```

File: _documentation/en/voice/voice-api/guides/signed-webhooks.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+With this limitation, you can use the signed webhooks now for a simple voice alert or two-factor authentication scenarios, using [Create an outbound call](https://developer.nexmo.com/api/voice#createCall) API request with embedded `ncco`:
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/signed-webhooks.md:134]:
```suggestion
With this limitation, you can use the signed webhooks now for a simple voice alert or two-factor authentication scenario, using a [Create an outbound call](https://developer.nexmo.com/api/voice#createCall) API request with an embedded `NCCO`:
```

File: _documentation/en/voice/voice-api/guides/signed-webhooks.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+  "ncco": [
+    {
+      "action": "talk",
+      "text": "Your code is '$TEMP_CODE'"
+    }
+  ]
+}
+```
+
+See also [Make an outbound call with an NCCO](/voice/voice-api/code-snippets/make-an-outbound-call-with-ncco) code snippet.
```
Comment (bencgreenberg) [inline _documentation/en/voice/voice-api/guides/signed-webhooks.md:157]:
```suggestion
See also the [Make an outbound call with an NCCO](/voice/voice-api/code-snippets/make-an-outbound-call-with-ncco) code snippet.
```

**Review by bencgreenberg — APPROVED** [review]

**Pull request merged**